KAA-1283: Review all C++ code samples in new Documentation

sparadiuk opened the pull request.

dr3amf4ll committed:
KAA-1157 - Delete explorer_icons.png

dr3amf4ll committed:
KAA-1157 - Add files via upload

dr3amf4ll committed:
KAA-1157 - Rest api images url fix

dr3amf4ll committed:
KAA-1157 - Add blockquote.note class

dr3amf4ll committed:
KAA-1157 - Update styles.css

dr3amf4ll committed:
KAA-1157 - Update styles.css

am1kk committed:
Add ability to choose key generation strategy.

Commit:
User key strategy small fix.
Add tests.

Commit:
Review fixes.

Commit:
Add Objective-C code and description to "Your First Kaa Application" guide.

Commit:
Review fixes, comment code.

Commit:
Review fixes.

Commit:
One more review fixes.

Commit:
Define upperTemperatureLimit and lowerTemperatureLimit as constants, fix if statement.

mlysenko committed:
KAA-676 Cross-compilation guide for C SDK (Intel Edison) on Linux platform.

RostakaGmfun committed:
KAA-1364: [C SDK] Fix undefined references when encryption is disabled

mlysenko committed:
KAA-676 Minor changes to the guide (after review).

sparadiuk committed:
Fixed all C++ code samples in new Documentation

Kirill380 committed:
CTL update of notifications

Kirill380 committed:
Notification documentation small fixes

Kirill380 committed:
CTL update of configuration documentation

Kirill380 committed:
CTL update of data collection documentation

Kirill380 committed:
Merge branch 'develop' into feature/KAA-1340

# Conflicts:
#	doc/Programming-guide/Key-platform-features/Data-collection/index.md

mlysenko committed:
Style: minor changes to document structure

Commit:
KAA-1165: Endpoint ownerhsip - initial commit.

Commit:
Fix temperature limit.

Commit:
KAA-1165: Endpoint ownership - reviewed.

Commit:
KAA-1165: Endpoint ownership - fixed upon AK review.

Kirill380 committed:
Merge branch 'develop' into feature/KAA-1340

# Conflicts:
#	doc/Programming-guide/Key-platform-features/Data-collection/index.md

rasendubi committed:
KAA-1407 Simplify git flow

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
```diff
@@ -377,8 +377,20 @@ When the implementation of your channel is ready you should add the channel to C
+    
+    ...
+    
+    kaa::IDataChannelPtr myChannel;
```
Is `myChannel` NULL?

sparadiuk replied:
fixed

rasendubi commented on doc/Programming-guide/Key-platform-features/Data-collection/index.md:
```diff
@@ -406,7 +404,7 @@ try {
     auto bucketInfo = recordInfo.getBucketInfo();
     std::cout << "Received log record delivery info. Bucket Id [" <<  bucketInfo.getBucketId() << "]. "
-              << "Record delivery time [" << recordInfo.getRecordDeliveryTimeMs() << " ms]." << std::endl;
+			  << "Record delivery time [" << recordInfo.getRecordDeliveryTimeMs() << " ms]." << std::endl;
```
don't use tabs

sparadiuk replied:
fixed

Kirill380 committed:
Changed links to ctl creation

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
```diff
@@ -609,9 +609,35 @@ To get an instance of KaaDataDemultiplexer for communication with the Bootstrap
 ### Step 5 - Push a response to the Kaa SDK
 
-The data demultiplexer contains only one method
+The KaaDataDemultiplexer contains only one method:
```
backticks

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
````diff
@@ -609,9 +609,35 @@ To get an instance of KaaDataDemultiplexer for communication with the Bootstrap
+
+</div><div id="cpp6" class="tab-pane fade" markdown="1" ><br/>
+
+```c_cpp
````
Does syntax highlight works correctly?

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
```diff
@@ -609,9 +609,35 @@ To get an instance of KaaDataDemultiplexer for communication with the Bootstrap
-The data demultiplexer contains only one method
+The KaaDataDemultiplexer contains only one method:
 
-[void processResponse(byte \[\] response)](http://kaaproject.github.io/kaa/doc/client-java-core/latest/org/kaaproject/kaa/client/channel/KaaDataDemultiplexer.html)
```
There was a reference to docs. Why was it removed?

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
````diff
@@ -424,7 +446,10 @@ To get an instance of [KaaDataMultiplexer](http://kaaproject.github.io/kaa/doc/c
 ```c_cpp
-    Kaa::getKaaClient().getOperationMultiplexer();
+
+    ...
````
not needed

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
````diff
@@ -439,7 +464,10 @@ To get an instance of KaaDataMultiplexer for communication with the Bootstrap se
 ```c_cpp
-    Kaa::getKaaClient().getBootstrapMultiplexer();
+
+    ...
````
not needed

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
````diff
@@ -553,7 +581,10 @@ To get an instance of [KaaDataDemultiplexer](http://kaaproject.github.io/kaa/doc
 ```c_cpp
-    Kaa::getKaaClient().getOperationDemultiplexer();
+
+    ...
````
not needed

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
````diff
@@ -568,16 +599,45 @@ To get an instance of KaaDataDemultiplexer for communication with the Bootstrap
 ```c_cpp
-    Kaa::getKaaClient().getBootstrapDemultiplexer();
+    
+    ...
````
not needed

rasendubi commented on doc/Customization-guide/Customizable-system-components/Communication-stack/index.md:
````diff
@@ -377,8 +377,30 @@ When the implementation of your channel is ready you should add the channel to C
 ```c_cpp
-    IDataChannelPtr myChannel;
-    Kaa::getKaaClient().getChannelManager().addChannel(myChannel);
+    #include <kaa/Kaa.hpp>
````
Remove extra indentation

Commit:
Fix default key generation strategy.

Commit:
Add documentation for new parameter.

Polinaa committed:
Changed jackson version in docs.

Pyshankov committed:
KAA-1293: Move email and user existence check methods from Web UI to REST API method

nocs00 committed:
KAA-1371: Update Programming guide -> Your first Kaa application

Pyshankov committed:
KAA-1431: Configuration parameter for default notification TTL

Pyshankov committed:
KAA-1431: Configuration parameter for default notification TTL

Pyshankov committed:
add applied changes to documentation

Pyshankov committed:
add applied changes to documentation

mlysenko committed:
Added a requirenment to build Boost and Avro for a host machine to cross-compilation section

mlysenko committed:
Fixed typos

vilimets committed:
KAA-1418 Obj-C Podspec updated. License updated (+1 squashed commit)
Squashed commits:
[cdeab11] KAA-1418 Obj-C project updated. Pod spec updated (+1 squashed commit)
Squashed commits:
[3643648] KAA-1418 Obj-C Pod spec added. Pods build fixes

mlysenko committed:
Fixed shell commands

Kirill380 committed:
Fix problem with log delivery time

Commit:
KAA-1436: Fixed REST for fetching credentials status

ikulikov committed:
Merge pull request #1065 from Polinaa/feature/KAA-1374

KAA-1374: Update a jackson version in docs

ikulikov committed:
Merge pull request #1079 from Kirill380/feature/KAA-1220

KAA-1220: [Java SDK] Incorrect calculation of log delivery time

ikulikov committed:
Merge pull request #1075 from nocs00/KAA-1371

KAA-1371: Update Programming guide -> Your first Kaa application

rasendubi committed:
KAA-1114 C SDK porting guide

mlysenko committed:
Fixed typos

vadimol committed:
Merge pull request #984 from mlysenko/KAA-676

KAA-676: Cross-compilation guide for C SDK (Intel Edison) on Linux

vadimol committed:
Merge pull request #1078 from mlysenko/KAA-1356

KAA-1356 Cross-compilation guide for C++ SDK (Intel Edison) on Linux platform

Kirill380 committed:
Merge branch 'develop' into feature/KAA-1340

# Conflicts:
#	doc/Programming-guide/Key-platform-features/Configuration-management/index.md
#	doc/Programming-guide/Key-platform-features/Notifications/index.md

vadimol committed:
Merge pull request #1082 from rasendubi/KAA-1114

KAA-1114 C SDK porting guide

Kirill380 committed:
Changed description about log schema creation

Kirill380 committed:
Small content fixes

Pyshankov committed:
Configuration parameter for default notification TTL

Kirill380 committed:
Fix with C++ event generation

Commit:
KAA-1419: Fix status file management logic

rasendubi committed:
Merge pull request #1086 from GamovCoder/KAA-1419

KAA-1419: Fix status file management logic

ikulikov committed:
Merge pull request #1085 from Kirill380/feature/KAA-1439

KAA-1439: Problem with generation event_class_family files

Commit:
KAA-1436: Remove unused variable in method KaaClient.getCredentialsStatus().

ikulikov committed:
Merge pull request #1022 from Kirill380/feature/KAA-1340

KAA-1340: Update documentation CTL

ikulikov committed:
Merge pull request #1081 from Rib47/kaa-1436

KAA-1436: Fix serever and client REST fuctionality for getting credentials status

Acarus committed:
Fixed C++ examples.

Commit:
KAA-1444: Method 'editUser' fixed.

vadimol committed:
KAA-1375 [C++ SDK] Failed to parse or compile schema: Input stream is not good

1) Added defaults schemas to avoid avrogen errors
2) Updated content of the KaaDefaults.cpp file
3) Added local gitignore file for C++ SDK
4) Added autogenereted files to gitignore

ikulikov committed:
Merge pull request #1089 from chv348/KAA-1444

KAA-1444: Fix method editUser() in AdminUI

vadimol committed:
Merge pull request #900 from sparadiuk/develop

KAA-1223 [C++] Your first Kaa application

hunch7 committed:
Merge pull request #1 from Acarus/kaa-1165-EPownership-review

KAA-1165: Endpoint ownership

Acarus committed:
Fixed uuids at rest api.

vadimol committed:
Merge pull request #1090 from vadimol/KAA-1375

KAA-1375 [C++ SDK] Failed to parse or compile schema

Pyshankov committed:
KAA-1293: Move email and user existence check methods from Web UI to REST API method

rasendubi committed:
Merge pull request #1084 from vilimets/KAA-1418

KAA-1418 [Objective-C] Change XCode project structure to use Podspec file

Pyshankov committed:
KAA-1293: Move email and user existence check methods from Web UI to REST API method

Commit:
KAA-1371: Your first Kaa application (updated with CTL info).

ikulikov committed:
Merge pull request #1069 from Pyshankov/KAA-1293

KAA-1293: Move email and user existence check methods from Web UI to …

Commit:
KAA-1340: Data collection - Minor rewording.

nocs00 committed:
KAA-1442: editUser() fixed sending email with temp password

nocs00 committed:
KAA-1442: configuration for mocking MessagingService

Commit:
KAA-1371: Your first Kaa application - fixed upon developer review.

Commit:
KAA-1340: Endpoint profiles - fixed upon developer review.

ikulikov committed:
Merge pull request #1097 from hunch7/kaa-1340-EPprofiles-CTLupdate

KAA-1340: Endpoint profiles - fixed upon developer review.

ikulikov committed:
Merge pull request #1095 from hunch7/kaa-1340-CTLupdate

KAA-1340:  Data collection - minor rewording.

ikulikov committed:
Merge pull request #1093 from hunch7/kaa-1371-kaaplication-CTLupdate

KAA-1371: Your first Kaa application (updated with CTL info).

Commit:
KAA-1168: Events - initial commit.

Commit:
KAA-1166: Configuration management - typo fixed.

ikulikov committed:
Merge pull request #1098 from hunch7/kaa-1166-configuration-review

KAA-1166: Configuration management - typo fixed.

ashvayka committed:
KAA-1310: Add validation of endpoint state while receiving events

nocs00 committed:
KAA-1401: added assisted attachment sample Java code

hunch7 committed:
Merge pull request #2 from nocs00/KAA-1401_review

KAA-1401: added assisted attachment sample Java code

Commit:
KAA-1168: Events - interim commit.

ashvayka committed:
KAA-1208: Same response comes again from Server

ikulikov committed:
Merge pull request #1099 from ashvayka/bug/KAA-1310

KAA-1310: App get event, when user isn't attached

Commit:
KAA-1135: Using Kaa endpoint SDKs - reviewed.

Acarus committed:
Fixed an inappropriate work with arrays in injectUuidsFromJsonNodes.

vadimol committed:
Merge pull request #1004 from RostakaGmfun/KAA-1364

KAA-1364: [C SDK] Fix undefined references when encryption is disabled

Commit:
KAA-1168: Events - interim commit.

ikulikov committed:
Merge pull request #1092 from Acarus/bug/KAA-1446

KAA-1446: Make field "__uuid" optional for configurations created from file (with AdminClient)

ashvayka committed:
KAA-1208: Test fix

Commit:
KAA-1168: Events - reviewed.

ikulikov committed:
Merge pull request #1100 from ashvayka/bug/KAA-1208

KAA-1208: Same response comes again from Server

vilimets committed:
KAA-1418 Obj-C CocoaLumberjack pod version specified

rasendubi committed:
Merge pull request #963 from am1kk/KAA-1222

KAA-1222: Programming guide -> Your first Kaa application (Objective-C code)

Commit:
KAA-1147: Using Kaa endpoint SDKs, C++ - reviewed.

Commit:
KAA-1147: Using endpoint SDKs, C++ - small fix (TOC added).

nocs00 committed:
KAA-1401: changed java sample for assisted attachment

vadimol committed:
Merge pull request #1103 from vilimets/KAA-1418

KAA-1418 Obj-C CocoaLumberjack pod version specified

hunch7 committed:
Merge pull request #3 from nocs00/KAA-1401_review2

KAA-1401: changed java sample for assisted attachment

sashadidukh committed:
KAA-1328: Update log appenders documentation according to data collection demo changes

rasendubi committed:
Merge pull request #1055 from rasendubi/KAA-1407

KAA-1407: Simplify git flow

rasendubi committed:
Merge pull request #902 from dr3amf4ll/patch-2

KAA-1157 - Add blockquote.note class

ikulikov committed:
Merge pull request #1102 from nocs00/KAA-1442

KAA-1442: REST method editUser do not send email with temp password

ikulikov committed:
Merge pull request #1106 from sashadidukh/bug/KAA-1328

KAA-1328: Update log appenders documentation according to data collection demo changes

ikulikov committed:
Merge pull request #1028 from hunch7/kaa-1165-EPownership-review

KAA-1165: Endpoint ownership - reviewed.

ikulikov committed:
Merge pull request #1105 from hunch7/kaa-1168-events-review

KAA-1168: Events - reviewed.

Pyshankov committed:
Configuration parameter for default notification TTL

ikulikov committed:
Merge pull request #1076 from Pyshankov/KAA-1431

KAA-1431: Configuration parameter for default notification TTL

vadimol committed:
Merge pull request #918 from am1kk/KAA-1234

KAA-1234: [Objective-C] Client authentication key pair generation build paramether

ikulikov committed:
Merge pull request #1101 from hunch7/kaa-1135-endpointSDKs-review

KAA-1135: Using Kaa endpoint SDKs - reviewed.

ikulikov committed:
Merge pull request #1104 from hunch7/kaa-1147-endpointSDKs-cpp

KAA-1147: Using Kaa endpoint SDKs, C++ - reviewed.

ashvayka committed:
KAA-1393: Remove unfinished low-priority documentation pages

Commit:
Documentation - Data collection - minor markup fix.

ashvayka committed:
KAA-1393: Removes incomplete low-priority documentation pages

ikulikov committed:
Merge pull request #1110 from hunch7/kaa-datacollection-fix

Documentation - Data collection - minor markup fix.

Commit:
Your first Kaa application - fixed upon QA review.

rasendubi committed:
Merge pull request #1111 from ashvayka/task/KAA-1393

KAA-1393: Remove unfinished low-priority documentation pages

ikulikov committed:
Merge pull request #1114 from hunch7/kaa-yourfirstapp-fixes

Your first Kaa application - fixed upon QA review.
